### DISCUSSION TITLE
How to persist the signin callbackurl when user is redirected to the error page

### DISCUSSION THREAD

**Discussion by madeinspace** [nextauthjs/next-auth#2593] [Help] [answered]
### Question 💬

I have an issue when ppl are entering the wrong credential, in the process of redirecting to the error page I lose the original callbackurl that the user is supposed to be redirected to after a successful signing.

It is actually the same if I set the error page to be the signing page, that callbackUrl is lost if users login is failing.

How can I persist that callback so the user can try again and be redirected to the original page on successful signing?

Any workaround you guys can think of? much appreciated thanks!

### How to reproduce ☕️

        async authorize(credentials, req) {
        console.log('credentials: ', credentials);

        // Add logic here to look up the user from the credentials supplied
        const user = await isAuthorised(credentials, req)

        // If no error and we have user data, return it
        if (user) {
          return user
        }
        // Return null if user data could not be retrieved
        // throw new Error('error signin in') // Redirect to error page
        // return null
        // throw '/path/to/redirect'        // Redirect to a URL

       // Is there a way here or elsewhere to keep the original callbackurl so the user can be redirected on success?

      }

### Contributing 🙌🏽

No, I am afraid I cannot help regarding this

**Comment by balazsorban44** [comment, marked as answer]
I would rather not redirect to another page on error, and improve the ux by showing a message to the user. Check https://next-auth.js.org/getting-started/client#using-the-redirect-false-option

**Reply by madeinspace** [reply]
unfortunately, I'm using SSR with a custom signiin page and that SignIn() function is client-side only apparently.

I'm thinking of storing the original callbackUrl in a cookie for retrieval if signing fail but that doesn't feel great haha

And yes ideally I'd prefer to stay on the signing page and display an error message there but I don't see how this is possible in my situation (SSR and custom signin)

**Reply by madeinspace** [reply]
EDIT: Actually that works, thanks for the heads up. I'll use the query to display an appropriate error message. Thanks heaps!
